extensions/wikia/ArticleVideo/scripts/article-video.js

```diff
@@ -0,0 +1,149 @@
+			return null;
+		}
+
+		function initVideo(onCreate) {
+			var ooyalaVideoId = window.wgArticleVideoData.videoId;
+
+			ooyalaVideoController = new OoyalaVideo(ooyalaVideoElementId, ooyalaVideoId, onCreate);
+		}
+
+		function uncollapseVideo() {
+			videoCollapsed = false;
+			$videoContainer.css({
+				'position': '',
+				'bottom': '',
+				'right': '',
+				'top': '',
+				'left': '',
+				'width': ''
+			});
+			$videoThumbnail.css('height', '');
+			$ooyalaVideo.css('height', '');
+			$video.removeClass('collapsed collapsed-ready');
+			if (ooyalaVideoController) {
+				updatePlayerControls(false);
+			}
+
+		}
+
+		function closeButtonClicked(event) {
+			event.stopPropagation();
+			if (ooyalaVideoController && ooyalaVideoController.player) {
+				ooyalaVideoController.player.pause();
+			}
+			uncollapseVideo();
+			collapsingDisabled = true;
+		}
+
+		function updatePlayerControls(waitForTransition) {
+			ooyalaVideoController.hideControls();
+			if (waitForTransition) {
+				$videoContainer.on(transitionEndEventName, function () {
+					ooyalaVideoController.sizeChanged();
+					ooyalaVideoController.showControls();
+				});
+			} else {
+				ooyalaVideoController.sizeChanged();
+				ooyalaVideoController.showControls();
+			}
+		}
+
+		function isVideoPlaying() {
+			if (ooyalaVideoController && ooyalaVideoController.player) {
+				return ooyalaVideoController.player.getState() === OO.STATE.PLAYING;
+			}
+		}
+
+		function isVideoInFullScreenMode() {
+			if (ooyalaVideoController && ooyalaVideoController.player) {
+				return ooyalaVideoController.player.getFullscreen();
+			}
+		}
+
+		function toggleCollapse() {
+			// That's for Safari because it triggers scroll event (it scrolls to the top)
+			// when video is switched to full screen mode.
+			if (isVideoInFullScreenMode()) {
+				return;
+			}
+			if (!collapsingDisabled || isVideoPlaying() || videoCollapsed) {
+				var scrollTop = $(window).scrollTop(),
+					videoHeight = $video.outerHeight(),
+					videoOffset = $video.offset(),
+					scrollOffset = 100,
+					collapseOffset = videoOffset.top + videoHeight - scrollOffset;
+
+				if (scrollTop > collapseOffset && !videoCollapsed) {
+					var videoWidth = $video.outerWidth(),
+						viewportWidth = $(window).width(),
+						viewportHeight = $(window).height();
+
+					collapsingDisabled = false;
+					videoCollapsed = true;
+					$video.addClass('collapsed-ready');
+					if (ooyalaVideoController) {
+						updatePlayerControls(true);
+					}
+					$videoContainer.css({
+						'bottom': viewportHeight - videoOffset.top - videoHeight + $(window).scrollTop(),
+						'right': viewportWidth - videoOffset.left - videoWidth,
+						'width': videoWidth
+					});
+					$videoThumbnail.css('height', videoHeight);
+					$ooyalaVideo.css('height', videoHeight);
+					setTimeout(function () {
+						if (videoCollapsed) { // we need to be sure video has not been uncollapsed yet
+							$video.addClass('collapsed');
+						}
+					}, 0);
```

Comment:
@WikiaTech is right, this is a complex function. How about extracting this block to `collapseVideo()`?